javascript/src/seedsource/containers/Upload.js

```diff
@@ -0,0 +1,210 @@
+import React from 'react'
+import { connect } from 'react-redux'
+import PropTypes from 'prop-types'
+import shp from 'shpjs'
+import { updateConstraintValues } from '../../actions/constraints'
+import { setError } from '../../actions/error'
+
+class Upload extends React.Component {
+    constructor(props) {
+        super(props)
+
+        // Would this be useful for reports?
+        // this.state = {
+        //     filename: '',
+        //     uploadMessages: {},
+        //     shapeType: ''
+        // }
+
+        this.handleFileUpload = this.handleFileUpload.bind(this)
+        this.handleUploadFinish = this.handleUploadFinish.bind(this)
+    }
+
+
+    handleFileUpload(event) {
+        event.stopPropagation()
+        event.preventDefault()
+
+        document.getElementById("loadShapefileModal").style.display = "inherit"
+
+        let files
+
+        if (event.dataTransfer && event.dataTransfer.files.length) {
+            files = event.dataTransfer.files
+        } else if (event.target) {
+            files = event.target.files
+        }
+
+        let zipFile
+        let shpFile
+        const prjFiles = {}
+        for (let i = 0; i < files.length; i += 1) {
```

Comment:
Use `.forEach` here, unless `files` is big enough to warrant performance concerns.